src/protocols/interaction_model/StatusCode.h

```diff
@@ -48,6 +50,108 @@ enum class Status : uint8_t
 const char * StatusName(Status status);
 #endif // CHIP_CONFIG_IM_STATUS_CODE_VERBOSE_FORMAT
 
+/**
+ * @brief Class to encapsulate a Status code, including possibly a
+ *        cluster-specific code for generic SUCCESS/FAILURE.
+ *
+ * This can be used everywhere a `Status` is used, but it is lossy
+ * to the cluster-specific code if used in place of `Status` when
+ * the cluster-specific code is set.
+ *
+ * This class can only be directly constructed from a `Status`. To
+ * attach a cluster-specific-code, please use the `ClusterSpecificFailure`
+ * and `ClusterSpecificSuccess` factory methods.
+ */
+class StatusCode
```

Comment:
I would consider calling this ResponseStatus or ClusterStatus; it's never really used for general status (e.g. on the IM level), right?

Reply:
This encapsulates what's needed to convey a StatusIB at the IM level in one value.

I want to add `CommandHandler::SetStatusCode` to replace many usages of `AddStatus`, `AddClusterSpecificSuccess`, `AddClusterSpecificFailure` with a single centralized call.

Reply:
I understand that.  My point is that on the IM level you never have cluster-specific codes...  There is some chance of confusion between InteractionModel::Status and InteractionModel::StatusCode and I am trying to figure out how to make it very clear which one should be used when.

Reply:
Renamed to ClusterStatus